### PR TITLE
Ventilator mode/oxygen modality data on consultation page

### LINKED ISSUE #8264: NIV and IV ventilator mode & Oxygen support Modality mode have no data reflection in patient consultation page
**Describe the bug**

In the patient consultation page, currently, there is a badge present their respiratory support is either NIV, IV, or OS but the related ventilator mode or modality mode is shown nowhere in the platform

image

**To Reproduce**

1. Go to patient consultation page
2. Click on log update and file the detailed log updated
3. post creation, go to the consultation page.

**Screenshots**
image
image

**Expected behavior**

- [ ] Add a table on the "ventilation tab" in patient dashboard (https://care-staging.ohc.network/facility/da95c77c-cf2f-4400-be53-6da766951552/patient/70ff1884-89ca-43fd-bc6c-2354cff9457d/consultation/9b1b3f6d-581d-46f3-8279-456596defd35/ventilator). The table to appear on top of the ventilator tab before the graphs of PIP and MAP
Screenshot 2024-08-19 at 7 38 30 PM

- [ ] Indicate change in modality and vantilator mode/oxygen modality by adding a vertical line as marker on the time series graphs for the various parameters
Screenshot 2024-08-19 at 7 40 20 PM

### LINKED ISSUE #8611: Change "Ventilator Spo2" to "Spo2" On Events Module
**Describe the bug**
When updating Spo2 values in the vital section on Brief update, Progress Note and Tele-Medicine Round Types, It shows as "Ventilator Spo2" on Events Module.
**To Reproduce**

1. Go to a patient file
2. Click on log Update and select Brief Update as round type
3. update the SPo2 field and submit, then go to events module 

**Expected behavior**

**Screenshots**
image

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### LINKED ISSUE #8612: Change the "Ventilator Interface" when Respiratory Support Selected as "Oxygen Support"
**Describe the bug**
When selecting and updating Oxygen Support as Respiratory Support on Detailed and Progress note log updates, The updated values are showing in the events module captioned as ventilator Spo2, Ventilator Oxygen Modality, Ventilator Interface as Oxygen Support and ventilator oxygen modality oxygen flow rate. 

(* Oxygen Support is not a Ventilator Parameter But a Respiratory Support type.)
 @aparnacoronasafe pls add suggestions
- [ ]  Change the caption "Ventilator Interface" When Oxygen Support is selected.
- [ ] Change the Ventilator Spo2 to Spo2
- [ ] Chante the Ventilator Oxygen Modality to Oxygen Modality
- [ ] Change the Ventilator Oxygen Modality Oxygen Flow Rate to Oxygen Flow Rate

**To Reproduce**

1. Go to a patient file and file a Detailed log update
2. update Respiratory Support parameter by selecting Support as Oxygen Support 
3.update oxygen Modality,oxygen flow rate and Spo2

**Expected behavior**

**Screenshots**
image

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by Jacobjeevan**
## Proposed Changes

- Fixes #8264 
- Fixes #8612 
- Fixes #8611 

image

Take a look at the font styling for the label, can adjust as needed; could do the same for label position as well (check the #8264 thread for examples).

image

image

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [x] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Release Notes

- **New Features**
  - Enhanced `ConsultationVentilatorTab` with data fetching, loading states, and pagination for ventilator reports.
  - Introduced `VentilatorTable` to display ventilator data with improved formatting and internationalization support.
  - Added vertical markers to `VentilatorPlot` for better data visualization.
  - New test suite for "Patient Doctor Connect" functionality, enhancing coverage of user interactions.
  
- **Improvements**
  - Expanded localization support for oxygen modalities and ventilator modes.
  - Improved error handling and data presentation in various components.

- **Bug Fixes**
  - Adjusted handling of undefined values in `BinaryChronologicalChart` component.

- **Chores**
  - Reorganized import statements across various test files for better readability.

### CODE REVIEW

**Comment by nihal467** [comment]
@Jacobjeevan is the PR ready for testing

**Comment by Jacobjeevan** [comment]
> @Jacobjeevan is the PR ready for testing

@nihal467 Yea, finalized on using the line graphs in the call yesterday. There maybe some changes with the label styling, but it's good to go for the most part.

Edit: Apart from any code review updates ofc.

**Comment by Jacobjeevan** [comment]
@aparnacoronasafe Do we need any additional changes here? As mentioned above, going with markLine graph as they are accurate when switching to bar graphs.

Edit: Changes made as requested, ready for review 👍

**Comment by samholics** [comment]
tested,
@nihal467 @Jacobjeevan 
- [ ] "Ventilator Spo2" is still showing up on events module when spo2 updated on vitals section on brief round type also on oxygen support section of Respiratory support. it should be renamed to "Spo2" on events module if its not on ventilator support.
- [ ] still oxygen support is showing under as "ventilator Interface" on events module
image
 note: oxygen support is not a Ventilator Interface

**Comment by Jacobjeevan** [comment]
> tested, @nihal467 @Jacobjeevan
> 
>     * [ ]  "Ventilator Spo2" is still showing up on events module when spo2 updated on vitals section on brief round type also on oxygen support section of Respiratory support. it should be renamed to "Spo2" on events module if its not on ventilator support.
> 
>     * [ ]  still oxygen support is showing under as "ventilator Interface" on events module
> 
> 
> image note: oxygen support is not a Ventilator Interface

I believe I missed the changes when merging the translations; I'll get it fixed 👍  

As far as this goes,

>  still oxygen support is showing under as "ventilator Interface" on events module

Is it acceptable to have "Respiratory support" as the caption for oxygen support as well as NIV/IV?  @samholics

**Comment by samholics** [comment]
> > tested, @nihal467 @Jacobjeevan
> > ```
> > * [ ]  "Ventilator Spo2" is still showing up on events module when spo2 updated on vitals section on brief round type also on oxygen support section of Respiratory support. it should be renamed to "Spo2" on events module if its not on ventilator support.
> > 
> > * [ ]  still oxygen support is showing under as "ventilator Interface" on events module
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > image note: oxygen support is not a Ventilator Interface
> 
> I believe I missed the changes when merging the translations; I'll get it fixed 👍
> 
> As far as this goes,
> 
> > still oxygen support is showing under as "ventilator Interface" on events module
> 
> Is it acceptable to have "Respiratory support" as the caption for oxygen support as well as NIV/IV? @samholics

yes we can do  " Respiratory Support Type"

**Comment by Jacobjeevan** [comment]
> yes we can do " Respiratory Support Type"

👍 

image

image

**Comment by samholics** [comment]
> > yes we can do " Respiratory Support Type"
> 
> 👍
> 
> image image

- [ ] Rename INVASIVE to INVASIVE VENTILATOR and Non-INVASIVE to NON-INVASIVE VENTILATOR on Respiratory support type
invasive or non-invasive alone doesn't make sense when its captioned under Respiratory Support type.

**Comment by Jacobjeevan** [comment]
>     * [ ]  Rename INVASIVE to INVASIVE VENTILATOR and Non-INVASIVE to NON-INVASIVE VENTILATOR on Respiratory support type
>       invasive or non-invasive alone doesn't make sense when its captioned under Respiratory Support type.

👍 

image

**Comment by samholics** [comment]
> > ```
> > * [ ]  Rename INVASIVE to INVASIVE VENTILATOR and Non-INVASIVE to NON-INVASIVE VENTILATOR on Respiratory support type
> >   invasive or non-invasive alone doesn't make sense when its captioned under Respiratory Support type.
> > ```
> 
> 👍
> 
> image

Tested good to go @nihal467

**Comment by nihal467** [comment]
image

@Jacobjeevan After I created a detailed log update, I submitted a brief log update for the patient. This resulted in the removal of the respiratory support badge from the patient consultation section. However, despite the removal of the respiratory badges, the ventilator log's end date and time were not updated. Can you explain how the ventilator end-time logic works?

**Comment by Jacobjeevan** [comment]
> image
> 
> @Jacobjeevan After I created a detailed log update, I submitted a brief log update for the patient. This resulted in the removal of the respiratory support badge from the patient consultation section. However, despite the removal of the respiratory badges, the ventilator log's end date and time were not updated. Can you explain how the ventilator end-time logic works?

End time is set to the `taken_at` of the next daily round, however I just checked and I'm filtering out rounds where `ventilator_interface` is null earlier in the code - which is causing this issue. I'll get it fixed 👍  Good catch.

**Comment by Jacobjeevan** [comment]
To offer an update, while working on Vitals redundant api call issue, I realized we could just use useDailyReports route at top level and pass in the data to both the table and VentilatorPlot (right now, VentilatorPlot uses DailyReportsAnalyse route while the table was using useDailyReports).

So finalizing those changes and testing right now 👍

**Comment by nihal467** [comment]
LGTM

File: src/components/Facility/ConsultationDetails/Events/EventsList.tsx [hunk 1]
```diff
@@ -64,6 +64,13 @@ export default function EventsList({ query }: { query: QueryParams }) {
                       }
 
                       const values = Object.fromEntries(entries);
+                      if (
+                        values.ventilator_interface &&
```
Comment (rithviknishad) [inline src/components/Facility/ConsultationDetails/Events/EventsList.tsx]: nitpicks: this is no-op right

```suggestion
```

File: src/components/Facility/Consultations/VentilatorTable.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  const { dailyRoundsList } = props;
+  let sortedData: DailyRoundsModel[] = [];
+
+  const sortData = (data: DailyRoundsModel[]) => {
+    return data.sort(function (a, b) {
+      const ad = new Date(a.taken_at ?? Date.now());
+      const bd = new Date(b.taken_at ?? Date.now());
+      return ad > bd ? 1 : -1;
+    });
+  };
```
Comment (rithviknishad) [inline src/components/Facility/Consultations/VentilatorTable.tsx]: checkout the utility fn: `compareBy` or `compareByDateString` right. we could re-use that here.

https://github.com/ohcnetwork/care_fe/blob/develop/src/Utils/utils.ts#L416-L428

File: src/components/Facility/ConsultationDetails/Events/EventsList.tsx (hunk 1 above)
Comment (Jacobjeevan) [reply src/components/Facility/ConsultationDetails/Events/EventsList.tsx]: Yea 😆

File: src/components/Facility/Consultations/VentilatorTable.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        ? round.ventilator_mode
+        : ventilatorInterface == "OXYGEN_SUPPORT"
+          ? round.ventilator_oxygen_modality
+          : null;
+    return modeOrModality;
+  };
+
+  const VentilatorTableBody = (dailyRoundsList: DailyRoundsModel[]) => {
+    const rows = [];
+    for (let index = 0; index < dailyRoundsList.length; index++) {
```
Comment (shivankacker) [inline src/components/Facility/Consultations/VentilatorTable.tsx:68]: a map should be more efficient and easy to use than a for loop here

File: src/components/Facility/Consultations/VentilatorPlot.tsx [hunk 4]
```diff
@@ -28,44 +26,194 @@ const modality: Array<ModalityType> = [
...
+        : ventilatorInterface == "OXYGEN_SUPPORT"
+          ? round.ventilator_oxygen_modality
+          : null;
+    return modeOrModality;
+  };
+
+  const getMarkLineData = (name: string) => {
+    const markLineData = [];
+    if (!dailyRoundsList) return [];
+    for (let index = 0; index < dailyRoundsList.length; index++) {
```
Comment (shivankacker) [inline src/components/Facility/Consultations/VentilatorPlot.tsx]: same here

**Review by shivankacker — CHANGES_REQUESTED** [review]

File: src/components/Facility/Consultations/VentilatorTable.tsx (hunk 3 above)
Comment (Jacobjeevan) [reply src/components/Facility/Consultations/VentilatorTable.tsx:68]: True, though within this loop, I'm traversing/skipping consecutive rounds (the while loop) that have same interface and mode/modality. That should be more readable with a for loop, no? 🤔  @shivankacker 

Same for below.

**Comment by bodhish** [comment]
@coderabbitai full review

**Comment by Jacobjeevan** [comment]
@coderabbitai full review

File: src/components/Facility/ConsultationDetails/ConsultationVentilatorTab.tsx [hunk 5]
```diff
@@ -1,21 +1,43 @@
...
+  const { consultationId } = props;
+  const { qParams, Pagination, resultsPerPage } = useFilters({ limit: 36 });
+
+  const { loading: isLoading, data } = useQuery(routes.getDailyReports, {
+    pathParams: { consultationId },
+    query: {
+      limit: resultsPerPage,
+      offset: (qParams.page ? qParams.page - 1 : 0) * resultsPerPage,
+    },
+  });
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Facility/ConsultationDetails/ConsultationVentilatorTab.tsx:22]: Not needed as children are handling the case where data is empty/undefined.

File: src/components/Facility/ConsultationDetails/Events/EventsList.tsx [hunk 6]
```diff
@@ -64,6 +64,12 @@ export default function EventsList({ query }: { query: QueryParams }) {
                       }
 
                       const values = Object.fromEntries(entries);
+                      if (
+                        values.ventilator_interface === "INVASIVE" ||
+                        values.ventilator_interface === "NON_INVASIVE"
+                      ) {
+                        values.ventilator_interface += " VENTILATOR";
+                      }
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Facility/ConsultationDetails/Events/EventsList.tsx:76]: This is related to displaying the values as per the requirement in daily logs and is needed.

File: src/components/Facility/Consultations/VentilatorPlot.tsx [hunk 7]
```diff
@@ -28,44 +26,194 @@ const modality: Array<ModalityType> = [
...
+  }, [dailyRoundsList]);
+
+  const getGraphData = (dailyRoundsData?: DailyRoundsModel[]) => {
+    let graphData = {};
+    const graphDataCount = dailyRoundsData?.length ?? 0;
+    if (dailyRoundsData) {
+      graphData = dailyRoundsData.reduce(
+        (acc, currentRound: DailyRoundsModel) => ({
+          ...acc,
+          //@ts-expect-error taken_at should always be available
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Facility/Consultations/VentilatorPlot.tsx]: Don't want to update the interface at the moment.

File: src/components/Facility/Consultations/VentilatorPlot.tsx [hunk 8]
```diff
@@ -28,44 +26,196 @@
...
+  const getModeOrModality = (round: DailyRoundsModel) => {
+    const ventilatorInterface = round.ventilator_interface;
+    const modeOrModality =
+      ventilatorInterface == "INVASIVE" || ventilatorInterface == "NON_INVASIVE"
+        ? round.ventilator_mode
+        : ventilatorInterface == "OXYGEN_SUPPORT"
+          ? round.ventilator_oxygen_modality
+          : null;
+    return modeOrModality;
+  };
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Facility/Consultations/VentilatorPlot.tsx]: Switching to case statements here.

File: src/components/Facility/Consultations/VentilatorPlot.tsx [hunk 9]
```diff
@@ -28,44 +26,196 @@
...
+        };
       });
-      if (res && res.ok && data) {
-        setResults(data.results);
-        setTotalCount(data.count);
-      }
-    };
+    }
+    return { graphData, graphDataCount };
+  };
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Facility/Consultations/VentilatorPlot.tsx]: Interface added for graph results.

File: src/components/Facility/Consultations/components/BinaryChronologicalChart.tsx [hunk 10]
```diff
@@ -3,7 +3,7 @@ import { formatDateTime } from "../../../../Utils/utils";
 
 export default function BinaryChronologicalChart(props: {
   data: {
-    value: boolean;
+    value: boolean | undefined;
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Facility/Consultations/components/BinaryChronologicalChart.tsx:7]: Not necessary as earlier checks would prevent such a scenario (additionally, timestamp should have values if there's daily rounds data).

File: src/components/Facility/Consultations/VentilatorTable.tsx (hunk 3 above)
Comment (shivankacker) [reply src/components/Facility/Consultations/VentilatorTable.tsx:68]: Sorry just saw this. If it seems more readable to you this way sure

**Review by shivankacker — APPROVED** [review]

File: src/components/Facility/ConsultationDetails/ConsultationVentilatorTab.tsx [hunk 11]
```diff
@@ -1,20 +1,44 @@
...
+import VentilatorTable from "@/components/Facility/Consultations/VentilatorTable";
+
+import useFilters from "@/hooks/useFilters";
+
+import routes from "@/Utils/request/api";
+import useQuery from "@/Utils/request/useQuery";
 
 export const ConsultationVentilatorTab = (props: ConsultationTabProps) => {
+  const { consultationId } = props;
+  const { qParams, Pagination, resultsPerPage } = useFilters({ limit: 36 });
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Facility/ConsultationDetails/ConsultationVentilatorTab.tsx:14]: It's 36 daily log events and not every event would have a corresponding ventilator table entry.

**Pull request merged**